coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/transformers/MapStructAccessOperatorTransformer.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.rel2trino.transformers;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlIdentifier;
+
+import com.linkedin.coral.com.google.common.collect.ImmutableList;
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+
+
+/**
+ * This class is an ad-hoc SqlCallTransformer which converts the map struct access operator "[]" defined
+ * from Calcite in a SqlIdentifier into a UDF operator of "element_at",
+ * e.g. from col["field"] to element_at(col, "field")
+ */
+public class MapStructAccessOperatorTransformer extends SqlCallTransformer {
+  private static final String AS_OPERATOR_NAME = "AS";
+  private static final Pattern MAP_STRUCT_ACCESS_PATTERN = Pattern.compile("\\\".+\\\"\\[\\\".+\\\"\\]");
+  private static final String ELEMENT_AT = "element_at(%s, %s)";
+
+  @Override
+  protected boolean condition(SqlCall sqlCall) {
+    if (AS_OPERATOR_NAME.equalsIgnoreCase(sqlCall.getOperator().getName())) {
```

Comment:
can do sqlCall.getOperator().kind == SqlKind.AS instead of string comparison

Reply:
This class is no longer needed.